README.md

```diff
@@ -1,234 +1,679 @@
+
+> [!IMPORTANT]
+> Note that:
+>
+> - The prediction time is neither the _time of the predicted event/label_ nor precisely the time _at which
+>   the prediction would be made in deployment_, but rather signifies the _inclusive endpoint_ of the data
+>   that can be used to predict the label for the given sample.
+> - The prediction time _may not correspond to an observed event time for the subject in the data._ This
+>   means you will need to generally use "as-of" joins to capture the relevant windows of time that are
+>   permissible (e.g., a join that matches to the last row with a timestamp less than or equal to the
+>   prediction time).
+
+In addition to the `subject_id` and `prediction_time`, the `Label` schema contains four additional, optional,
+_not nullable_ columns that capture the different kinds of labels that can be observed.
+Note that these columns are optional (they may or may not be present) but _not_ nullable (if they are present,
+they should have uniformly observed values). Only one of these columns should generally be populated:
+
+1. `boolean_value`: The label for a boolean task.
+2. `integer_value`: The label for tasks taking on integral values (e.g., ordinal tasks or integral numeric
+    tasks).
+3. `float_value`: The label for tasks taking on float values (e.g., continuous numeric tasks)
+4. `categorical_value`: The label for tasks taking on categorical values.
+
+#### Examples:
+
+```python
+>>> from meds import Label
+>>> Label.schema()
+subject_id: int64
+prediction_time: timestamp[us]
+boolean_value: bool
+integer_value: int64
+float_value: float
+categorical_value: string
+>>> Label.validate(pa.Table.from_pydict({
+...     "subject_id": [1, 2, 3],
+...     "prediction_time": [datetime(2021, 3, 1), datetime(2021, 4, 1), datetime(2021, 5, 1)],
+...     "boolean_value": [True, False, False],
+... })) # No issue
+>>> Label.validate(pa.Table.from_pydict({
+...     "subject_id": [1, 2, 3],
+...     "prediction_time": [datetime(2021, 3, 1), datetime(2021, 4, 1), datetime(2021, 5, 1)],
+...     "categorical_value": ["high", None, "low"],
+... }))
+Traceback (most recent call last):
+    ...
+flexible_schema.exceptions.TableValidationError:
+    Columns that should have no nulls but do: categorical_value
+
+```
+
+## Organization on Disk
+
+A MEDS dataset is organized under a root directory (`$MEDS_ROOT`) into several subfolders corresponding to the
+different schema components. Below is an overview of the directory structure:
+
+- **`$MEDS_ROOT/data/*`**
+
+    - Contains the event data files following the `Data` schema.
+    - Data is stored as a series of (possibly nested) sharded DataFrames in `parquet` format.
+    - The file glob `glob("$MEDS_ROOT/data/**/*.parquet")` will match all these sharded files.
+    - Each file holds all events for one or more subjects, each of which is sorted by time.
+
+- **`$MEDS_ROOT/metadata/dataset.json`**
```

Comment:
I would include MEDS in the suggested path/filenames somewhere, for similar reason to including MEDS in the class names. Otherwises there is no immediate message to the user that this is a MEDS data package.